### PR TITLE
Leanplum filter in-app message

### PR DESCRIPTION
**Pull request by vincent-re**
Implementation of in-app message filtering  leanplum-javascript-sdk
allow to filter message on :
 - withTriggers
 - withLimits
 - withLimitTimes

changes : 
- ActionManager class ( handle filtering )
- VarCache update : allow to save and retrieve in localstrorage the amount of messages views 

test:

task :
https://teamiflix.atlassian.net/browse/MES-66

### CODE REVIEW

File: src/ActionManager.js [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+import VarCache from "./VarCache";
+
+/** private
+ * return true if message pass one trigger/verb condition
+ * @param params {Object}
+ * @param message
+ * @returns {Boolean}
+ */
+const filterByTriggers = (params={}) => (message) => {
+  // TODO  check if exist, maybe define trigger locally, check then
```
Comment (voslartomas) [inline src/ActionManager.js]: What about those todos?

File: src/Constants.js [hunk 2]
```diff
@@ -104,10 +104,52 @@ export default {
...
+    PUSH_SUBSCRIPTION: '__leanplum_push_subscription',
+    MESSAGE_VIEW: '__leanplum_messages_view',
   },
 
   VALUES: {
     DETECT: '(detect)'
-  }
+  },
+
+  // TRIGGERS_VERBS: {
```
Comment (voslartomas) [inline src/Constants.js]: Why is all this commented out?

File: src/VarCache.js [hunk 3]
```diff
@@ -73,6 +75,52 @@ export default class VarCache {
     LocalStorageManager.saveToLocalStorage(Constants.DEFAULT_KEYS.TOKEN, VarCache.token)
   }
 
+  /**
+   * record view message in local storage and  in memory (aka session)
+   * @param messageId
+   */
+  static addMessageView(messageId){
+    // TODO : Check messageID
```
Comment (voslartomas) [inline src/VarCache.js]: Some TODO to do...

**Review by voslartomas — CHANGES_REQUESTED** [review]
You should also bump version of the whole library. There is a special script for that in the root.

File: src/ActionManager.js (hunk 1 above)
Comment (vincent-re) [reply src/ActionManager.js]: removed, thx

File: src/VarCache.js (hunk 3 above)
Comment (vincent-re) [reply src/VarCache.js]: fixed

**Review by voslartomas — APPROVED** [review]
Seems good. Don't forget to npm publish

**Pull request merged**